src/core/crypto/relay_crypto.c

```diff
@@ -142,6 +142,13 @@ relay_decrypt_cell(circuit_t *circ, cell_t *cell,
           if (relay_digest_matches(thishop->crypto.b_digest, cell)) {
             *recognized = 1;
             *layer_hint = thishop;
+            /* Keep current digest of this cell for the possible SENDME. */
+            if (thishop->crypto.sendme_digest) {
+              crypto_digest_free(thishop->crypto.sendme_digest);
```

Comment:
Should we log a BUG warning if we drop this value without having sent it in a SENDME cell?

Reply:
Commit 3943d096cc060d7676e1549e37d9577951b3b68f now makes it that we overwrite... At this point, we do not know if we are dropping it or not :S.